bintool/bintool.cpp

```diff
@@ -546,6 +546,11 @@ uint32_t calc_checksum(std::vector<uint8_t> bin) {
 
 #if HAS_MBEDTLS
 void hash_andor_sign_block(block *new_block, const public_t public_key, const private_t private_key, bool hash_value, bool sign, std::vector<uint8_t> to_hash) {
+    if (!(hash_value || sign)) {
```

Comment:
If neither `hash_value` nor `sign` are set, then perhaps it'd make more sense to just not call `hash_andor_sign_block` at all, rather than calling it and early-exiting? :shrug:

Reply:
That requires duplicate code in the 3 places it's called, so I think this is neater - context is that the `hash_andor_sign` functions for BINs and ELFs now also have a valid use case of adding a load_map without any hashing/signing (eg for a packaged SRAM binary), so the `hash_andor_sign` functions can be used to do both, either, or neither

Reply:
"so I think this is neater" - OK :)